prac_05/word_occurences.py

```diff
@@ -0,0 +1,19 @@
+"""Program to count occurrences of words in a string and return list of words with their count."""
+
+
+def main():
+    """Program to count occurrences of words in a string."""
+    sentence = input("Enter sentence: ").split()
+    sentence.sort()
+    sentence_dict = {}
+    for word in sentence:
+        if word in sentence_dict:
+            sentence_dict[word] += 1
+        else:
+            sentence_dict[word] = 1
+    for word, count in sentence_dict.items():
+        max_length = max((len(word) for word in sentence))
```

Comment:
Why calculate this inside the loop? It's the same every time. So, if you have 100 words, you calculate the same maximum 99 times more than you need to.